### DISCUSSION TITLE
keycloak-js change password flow

### DISCUSSION THREAD

**Discussion by migue99angel** [keycloak/keycloak#20968] [Q&A] [answered]
Hi,

I use the keycloak-js package to integrate a keycloak server with my javascript application. I wonder if there is any way to start the change password flow by using the adapter.

In the interface, I just see the following methods

```js
export interface KeycloakAdapter {
	login(options?: KeycloakLoginOptions): KeycloakPromise<void, void>;
	logout(options?: KeycloakLogoutOptions): KeycloakPromise<void, void>;
	register(options?: KeycloakRegisterOptions): KeycloakPromise<void, void>;
	accountManagement(): KeycloakPromise<void, void>;
	redirectUri(options: { redirectUri: string; }, encodeHash: boolean): string;
}
```
It doesn't seem to have a method with the reset password purpose. Is there any parameter for one of those methods which could allow me to redirect to the keycloak reset password view?

**Comment by jonkoops** [comment, marked as answer]
From the top of my head I believe you can trigger this from the login flow as folllows:

```js
keycloak.login({
  action: "UPDATE_PASSWORD",
});
```

**Reply by jonkoops** [reply]
See: https://www.keycloak.org/docs/latest/securing_apps/#methods

**Reply by migue99angel** [reply]
Thank you!
